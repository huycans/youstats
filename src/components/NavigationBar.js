import React from "react";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";
import Container from "react-bootstrap/Container";
import { GoogleLogout, GoogleLogin } from "react-google-login";

import youtubeIcon from "../static/images/youtube-icon.png";

export default function NavigationBar({
  signedin,
  handleSignout,
  handleSignin,
  setLoading
}) {
  return (
    <header id="header" className="header container" data-testid="app-header">
      <Navbar expand="md" bg="light" variant="light" data-testid="navbar">
        <Container>
          <Navbar.Brand data-testid="navbar-brand" href="/">
            <img
              data-testid="navbar-brand-img"
              alt="Youtube logo"
              src={youtubeIcon}
              width="40"
              height="30"
              className="d-inline-block align-top"
            />{" "}
            Stats
          </Navbar.Brand>
          <Navbar.Toggle aria-controls="responsive-navbar-nav" />
          <Navbar.Collapse
            className="justify-content-end"
            id="responsive-navbar-nav"
            data-testid="navbar-collapse-content"
          >
            <Nav.Item>
              {signedin ? (
                <GoogleLogout
                  clientId={process.env.REACT_APP_CLIENTID}
                  buttonText="Sign out of Google"
                  onLogoutSuccess={handleSignout}
                  render={(renderProps) => (
                    <button
                      data-testid="navbar-item-logout-btn"
                      className="auth-btn"
                      onClick={renderProps.onClick}
                      disabled={renderProps.disabled}
                    >
                      Sign out
                    </button>
                  )}
                ></GoogleLogout>
              ) : (
                <GoogleLogin
                  data-testid="google-login-btn"
                  clientId={process.env.REACT_APP_CLIENTID}
                  buttonText="Sign in with Google"
                  onSuccess={handleSignin}
                  onFailure={handleSignin}
                  cookiePolicy={"single_host_origin"}
                  isSignedIn={true}
                  onAutoLoadFinished={() => setLoading(false)}
                  render={(renderProps) => (
                    <button
                      data-testid="navbar-item-login-btn"
                      className="auth-btn"
                      onClick={renderProps.onClick}
                      disabled={renderProps.disabled}
                    >
                      Sign in with Google
                    </button>
                  )}
                />
              )}
            </Nav.Item>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    </header>
  );
}
