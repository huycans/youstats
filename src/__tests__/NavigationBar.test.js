import React from "react";
import renderer from "react-test-renderer";
import NavigationBar from "../components/NavigationBar";

import { render, cleanup, fireEvent } from "@testing-library/react";

//module react-google-login is being mocked using jest

const signedin = true;
const handleSignout = jest.fn();
describe("<NavigationBar />", () => {
  beforeEach(() => {
    cleanup();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <NavigationBar signedin={signedin} handleSignout={handleSignout} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render the logo", () => {
    const { queryByTestId } = render(
      <NavigationBar signedin={signedin} handleSignout={handleSignout} />
    );
    expect(queryByTestId("navbar-brand-img")).toBeTruthy();
  });

  it("does not render the logout btn when not signed in", () => {
    const { queryByTestId } = render(
      <NavigationBar signedin={false} handleSignout={handleSignout} />
    );
    expect(queryByTestId("navbar-item-logout-btn")).toBeFalsy();
  });

  it("render the logout btn when signed in", () => {
    const { queryByTestId } = render(
      <NavigationBar signedin={signedin} handleSignout={handleSignout} />
    );
    expect(queryByTestId("navbar-item-logout-btn")).toBeTruthy();
    expect(queryByTestId("navbar-item-logout-btn")).toHaveTextContent(
      "Sign out"
    );
  });
});
