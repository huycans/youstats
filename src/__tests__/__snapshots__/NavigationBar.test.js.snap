// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar /> renders correctly 1`] = `
<header
  className="App-header container"
  data-testid="app-header"
>
  <nav
    className="navbar navbar-expand-md navbar-light bg-light"
    data-testid="navbar"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        data-testid="navbar-brand"
        href="/"
      >
        <img
          alt="Youtube logo"
          className="d-inline-block align-top"
          data-testid="navbar-brand-img"
          height="30"
          src="youtube-icon.png"
          width="40"
        />
         
        Stats
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="justify-content-end navbar-collapse collapse"
        data-testid="navbar-collapse-content"
        id="responsive-navbar-nav"
      >
        <div
          className="nav-item"
        >
          <button
            className="signout-btn"
            data-testid="navbar-item-logout-btn"
            disabled={true}
            onClick={[Function]}
          >
            Sign out
          </button>
        </div>
      </div>
    </div>
  </nav>
</header>
`;
